numpy/lib/utils.py

```diff
@@ -1123,4 +1133,48 @@ def _opt_info():
             enabled_features += f" {feature}?"
 
     return enabled_features
+
+
+def _descr_without_metadata(descr):
+    l = list()
+    for d in descr:
+        d = list(d)
+        if isinstance(d[1], tuple):
+            d[1] = d[1][0]
+        elif isinstance(d[1], list):
+            d[1] = _descr_without_metadata(d[1])
+        l.append(tuple(d))
+    return l
+
+
+def dtype_without_metadata(dt):
+    """
+    Returns a dtype which is the image of the one passed in parameter and 
+    which, as well as its underlying dtypes, contains no metadata
+
+    Parameters
+    ----------
+    dt : dtype
+        Original dtype that may contain metadata
+
+    Returns
+    -------
+    result : dtype
+        Dtype containing no metadata
+    """
+    descr = dt.descr
```

Comment:
We could do this in python as a utility, implementing it as `dtype.drop_metadata()` or so would have the advantage of easier generalizing user DTypes.

Fields *can* be nested, so I think this must use a recursive approach.  I.e. check `dtype.names is None` if it is we should iterate `dtype.fields`, if not check `dtype.shape != ()` (not a subarray).  After that, we would have to assume `dtype.str` works (for now).

Both structured (names and shape) will  need to then call the function again recursively.

Reply:
ok but this means that drop_metadata is a method of dtype class, right ? 
Is there a python file where we can put this method in numpy project ? or we have to code it in cython. I'm not confortable with this language

Reply:
If we are OK with private, it could probably stay as a function for now.  If not, I think I like a method?  Even then it could be implemented in Python (for most of the logic at least), we would need a C stub that forwards it to Python.  That should already be done for some other methods or getters/setters.

But if that is what we do, and you are not happy to dig into Python C-API a bit on your own, I can help with that part.

Reply:
there something that i do not understand. You suggest `dtype.drop_metadata()` with no parameter, so i suppose it is a method of dtype class. 
And as metadata is a readonly attribute, `drop_metadata(self)` must return a copy, no?. 
Or you want to add a setter to metadata in order to allow `self`  to clean all  its metadatas ? In this case `drop_metadata(self)` returns nothing

Reply:
Yes, it would have to return a new dtype.  Honestly, dtypes should be immutable anyway.  I suppose if there is no metadata to drop, returning the same one would be fine/good.